### PR TITLE
feat(MeetingsJsonAdapter): add member-roster control

### PR DESCRIPTION
**Pull request by alinasuciu**
Add member roster control to give the possibility to show/hide roster of a given meeting id and create the unit tests accordingly  

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-225208

### CODE REVIEW

File: src/components/WebexMeeting/WebexMeeting.jsx [hunk 1]
```diff
@@ -74,6 +74,6 @@ WebexMeeting.defaultProps = {
   // eslint-disable-next-line no-confusing-arrow
   controls: (isActive) =>
     isActive
-      ? ['mute-audio', 'mute-video', 'share-screen', 'leave-meeting']
+      ? ['mute-audio', 'mute-video', 'share-screen', 'leave-meeting', 'member-roster']
```
Comment (maloun96) [inline src/components/WebexMeeting/WebexMeeting.jsx:77]: Here should be `participant-list`. It's an icon. https://momentum-ui.com/icons

Comment (alinasuciu) [reply src/components/WebexMeeting/WebexMeeting.jsx:77]: This array contains the names of the controls available in adapter not the names of the icons, the icon name is added inside de control

File: src/adapters/MeetingsJSONAdapter.js [hunk 2]
```diff
@@ -204,12 +212,21 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
       filter((event) => event.detail.ID === ID),
       tap(() => end$.next(`Meeting "${ID}" has completed.`)),
     );
-
-    const events$ = merge(audioEvents$, videoEvents$, joinEvents$, leaveEvents$, shareEvents$).pipe(
-      filter((event) => event.detail.ID === ID),
-      // Make a copy of the meeting to treat it as if were immutable
-      map((event) => ({...event.detail})),
-    );
+    const rosterEvent$ = fromEvent(document, ROSTER_CONTROL);
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:215]: while I prefer the singular in this case (each message emitted by this observable is a single event), let's use a plural here, for consistency with the other observables above.

File: src/adapters/MeetingsJSONAdapter.js [hunk 3]
```diff
@@ -581,6 +598,64 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
     return concat(default$, shareEvent$);
   }
 
+  /**
+   * Handles the toggle of the roster.
+   *
+   * @param {string} ID ID of the meeting for which to update display
+   * @private
+   */
+  async handleRoster(ID) {
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:607]: to align with the other controls' display function names, let's call this `rosterControl`

File: src/adapters/MeetingsJSONAdapter.js [hunk 4]
```diff
@@ -581,6 +598,64 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
...
+    };
+    const inactive = {
+      ID: ROSTER_CONTROL,
+      icon: 'participant-list',
+      tooltip: 'Show participants panel',
+      state: MeetingControlState.ACTIVE,
+      text: 'Participants',
+    };
+
+    const initialState$ = new Observable((observer) => {
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:638]: try subscribing the behavior subject to the stream of events:
```js
if (meeting) {
    state$ = new BehaviorSubject(showRoster ? active : inactive);
    events$ = fromEvent(....).pipe(...);
    events$.subscribe(state$);
} else {
    state$ = throwError(...);
}

return state$;
```

File: src/data/meetings.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ export default {
     remoteAudio: new MediaStream(),
     remoteVideo: new MediaStream(),
     remoteShare: new MediaStream(),
+    showRoster: false,
```
Comment (cipak) [inline src/data/meetings.js:11]: can we have a meeting with `showRoster: true`?

Comment (cipak) [reply src/data/meetings.js:11]: sorry, ignore this, I just saw that `meeting5` has its roster shown

**Comment by alinasuciu** [comment]
Add member-roster in MeetingJsonAdapter

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-225208

**Review by lalli-flores — APPROVED** [review]

**Pull request merged**